apis/kueue/v1alpha1/dynamicresourceallocationconfig_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+
+// DynamicResourceAllocationConfig is a singleton CRD that maps a logical resource name to one or more DeviceClasses
+// in the cluster. Only one instance named "default" in the kueue-system namespace
```

Comment:
I mentioned this elsewhere but we would really need the namespace to be configurable.

When users install kueue they are free to choose the namespace to put these resources. Hard coding kueue-system will break us.

Reply:
Once this validations are implemented, we want to implement tests in test/integration/singlecluster/webhook/core.